src/shared/src/backend-supports/project-supports.ts

```diff
@@ -90,11 +91,29 @@ const calculateProjectStartDate = (wps: { duration: number; startDate: Date }[])
   return minDate;
 };
 
+// calculate the project's status based on its pacakges' status
+const calculateProjectStatus = (proj: { workPackages: { wbsElement: { status: WBS_Element_Status } }[] }) => {
+  let isActive = false;
+  let isComplete = true;
+
+  if (proj.workPackages.length === 0) {
+    return WbsElementStatus.Inactive;
+  }
+
+  proj.workPackages.forEach((wp) => {
+    isComplete = isComplete && wp.wbsElement.status === WbsElementStatus.Complete;
+    isActive = isActive || wp.wbsElement.status === WbsElementStatus.Active;
+  });
+
+  return isComplete ? WbsElementStatus.Complete : isActive ? WbsElementStatus.Active : WbsElementStatus.Inactive;
```

Comment:
I don't love this in a nested ternary, I'd add some parentheses, or put it in an if statement personally